templates/common/fs.go

```diff
@@ -109,6 +111,14 @@ type CopyParams struct {
 	// visitor is an optional function that will be called for each file in the
 	// source, to allow customization of the copy operation on a per-file basis.
 	Visitor CopyVisitor
+
+	// If Hasher and OutHashes are not nil, then each copied file will be hashed
```

Comment:
What are the situations in which we expect them to be nil?

Reply:
We use copyRecursive in a few places as a general directory copying tool: (1) copying files from the template dir to the scratch dir in an include action (2) copying the temp dir to the output dir in goldentest recording (3) committing the scratch dir to the destination dir. We only want hashing for case 3.